frontend/packages/db-structure/src/schema/dbStructure.ts

```diff
@@ -34,39 +36,85 @@ export type Index = v.InferOutput<typeof indexSchema>
 const indicesSchema = v.record(indexNameSchema, indexSchema)
 export type Indices = v.InferOutput<typeof indicesSchema>
 
-export const tableSchema = v.object({
-  name: tableNameSchema,
-  columns: columnsSchema,
-  comment: v.nullable(v.string()),
-  indices: indicesSchema,
-})
-
-export type Table = v.InferOutput<typeof tableSchema>
-
-const cardinalitySchema = v.picklist(['ONE_TO_ONE', 'ONE_TO_MANY'])
-export type Cardinality = v.InferOutput<typeof cardinalitySchema>
-
-const foreignKeyConstraintSchema = v.picklist([
+const foreignKeyConstraintReferenceOptionSchema = v.picklist([
   'CASCADE',
   'RESTRICT',
   'SET_NULL',
   'SET_DEFAULT',
   'NO_ACTION',
 ])
+export type ForeignKeyConstraintReferenceOption = v.InferOutput<
+  typeof foreignKeyConstraintReferenceOptionSchema
+>
 
+const primaryKeyConstraintSchema = v.object({
+  type: v.literal('PRIMARY KEY'),
+  name: constraintNameSchema,
+  columnNames: v.array(columnNameSchema),
```

Comment:
You might not be considering adopting a composite key. If not, I'm planning to modify it like the follows and fix the related codes:

```suggestion
  columnName: columnNameSchema,
```

Reply:
We want to support composite keys because we want to accurately represent the state of our database.
However, I also feel that this is a feature that is not used very often. Please let me know if support would greatly complicate the implementation and experience.

Reply:
> Please let me know if support would greatly complicate the implementation and experience.

If the Liam supports composite keys, I think the `relationships` should reflect that as well.
Maybe we can discuss introducing the composite key feature as a separate topic.

I'll revert this change for now 😉

Reply:
I see, Thanks 👍🏻